bestiary/core/schema.py

```diff
@@ -246,9 +267,77 @@ class Meta:
     )
 
 
+class EcosystemPaginatedType(AbstractPaginatedType):
+    entities = graphene.List(EcosystemType)
+    page_info = graphene.Field(PaginationType)
+
+
+class AddEcosystem(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        title = graphene.String(required=False)
+        description = graphene.String(required=False)
+
+    ecosystem = graphene.Field(lambda: EcosystemType)
+
+    @check_auth
+    def mutate(self, info, name, title=None, description=None):
+        user = info.context.user
+        ctx = BestiaryContext(user)
+
+        ecosystem = add_ecosystem(ctx, name, title=title, description=description)
+
+        return AddEcosystem(
+            ecosystem=ecosystem
+        )
+
+
+class UpdateEcosystem(graphene.Mutation):
+    class Arguments:
+        ecosystem_id = graphene.Int()
+        data = EcosystemInputType()
+
+    ecosystem = graphene.Field(lambda: EcosystemType)
+
+    @check_auth
+    def mutate(self, info, ecosystem_id, data):
+        user = info.context.user
+        ctx = BestiaryContext(user)
+
+        ecosystem = update_ecosystem(ctx, ecosystem_id, **data)
+
+        return UpdateEcosystem(
+            ecosystem=ecosystem
+        )
+
+
+class DeleteEcosystem(graphene.Mutation):
+    class Arguments:
+        ecosystem_id = graphene.Int()
+
+    ecosystemName = graphene.String()
+
+    @check_auth
+    def mutate(self, info, ecosystem_id):
```

Comment:
`ecosystem_id` is a redundant name. It should be `id` because the operation is related to ecosystems, not to anything else.